src/shell/CMakeLists.txt

```diff
@@ -46,8 +46,14 @@ add_test(lean_ghash2   "${CMAKE_BINARY_DIR}/bin/lean" --githash)
 add_test(lean_unknown_option bash "${LEAN_SOURCE_DIR}/cmake/check_failure.sh" "${CMAKE_BINARY_DIR}/bin/lean" "-z")
 add_test(lean_unknown_file1 bash "${LEAN_SOURCE_DIR}/cmake/check_failure.sh" "${CMAKE_BINARY_DIR}/bin/lean" "boofoo.lean")
 
-# LEANC_OPTS is necessary for macOS c++ to find its headers
-set(TEST_VARS "PATH=${LEAN_BIN}:$PATH ${LEAN_TEST_VARS} CXX='${CMAKE_CXX_COMPILER} ${LEANC_OPTS}'")
+if(${EMSCRIPTEN})
+  set(LEAN_EXE "node --stack-size=8192 ${CMAKE_BINARY_DIR}/bin/lean")
```

Comment:
What exactly is `lean` and how does it relate to `lean.wasm`?

Reply:
`lean` is a javascript file. It loads `lean.wasm`.

We could rename `lean` to `lean.js`  by removing `unset(CMAKE_EXECUTABLE_SUFFIX)` in `src/CMakeLists.txt` if you prefer. However, this causes an issue with the line `"${PREV_STAGE}/bin/lean${CMAKE_EXECUTABLE_SUFFIX}"` in `stdlib.make.in` because our stage0 does not have the suffix `.js`. We would need something like `${PREV_STAGE_CMAKE_EXECUTABLE_SUFFIX}` or so.

Reply:
Why is the JS file so big, is it all the exported symbols? `PREV_STAGE_CMAKE_EXECUTABLE_SUFFIX` doesn't sound too weird to me. The upside I assume would be that we could replace `LEAN_EXE` with a `lean` shell script that does the `node` invocation. Not sure. What do you think?

Reply:
Yes, but I think we need to export all symbols because the interpreter can't find them otherwise.

I have added `PREV_STAGE_CMAKE_EXECUTABLE_SUFFIX` and replaced `LEAN_EXE` by a script.